File: RentEase.Domain/Models/Entities/Item.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using RentEase.Domain.Enums;
using RentEase.Domain.Models.Abstractions;

namespace RentEase.Domain.Models.Entities;

public class Item : BaseEntity
{
    public string Title { get; set; }
    
    public string Description { get; set; }
    
    public Category Category { get; set; }
    
    public int PriceUS { get; set; }

    [NotMapped]
    public int PriceUA => PriceUS * 36;
    
    public string Address { get; set; }
    
    public string ImageUrl { get; set; }
    
    // public string Longitude { get; set; }
    //
    // public string Latitude { get; set; }
    
    public string LandlordId { get; set; } 
    
    public bool IsAvailable { get; set; }
    
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public User Landlord { get; set; }
    
    public List<Order> Orders { get; set; }
}

File: RentEase.Mobile/rent_ease_mobile/src/shared/models/item.ts
import { Category } from "../enums/category.enum";
import { Order } from "./order";
import { User } from "./user";

export class Item {
    id:number = 0;
    title:string = '';
    description:string = '';
    category:Category = 0;
    priceUS:number = 0;
    priceUA?:number = 0;
    address:string = '';
    imageUrl:string = '';
    isAvaiable:boolean = false;
    landlordId:string = '';
    landlord?:User = new User();
    orders?:Order[];
}